test/fixtures/import-as-es6-extension/background.js

```diff
@@ -0,0 +1,3 @@
+browser.runtime.onMessage.addListener(async (msg, sender) => {
```

Comment:
The `browser` here is the global `browser` object, which is not the same as the one from the module (unless the module replaced the `browser` global).

If you want to verify that module loading works as expected, then `globalThis.browser` should be undefined (in Chrome) and defined in Firefox, and the file should start with `import {browser} from './browser-polyfill.js';`

Can you confirm that this (i.e. module script replacing the `browser` global instead of only exporting a browser object) is the intended behavior?

Reply:
By looking to what the umd wrapper is doing (there are fragments of its content in the PR description), you may notice that it is passing as the `global` parameter the result of `typeof globalThis !== "undefined" ? globalThis : typeof self !== "undefined" ? self : this` and so where globalThis is defined (and the script isn't loaded as an AMD or CommonJS module) the result is that it is going to call `globalThis.browser = mod.exports`.

And so, besides exporting browser from the ES6 module, it is also exporting it as a global.
Based on what #202 describes and the fix applied upstream in babel/babel#10477, this seems to be exactly the intended behavior.

Currently the test case is mainly a smoke test for #202 expectations, but testing explicitly that `import ... from './browser-polyfill.js';` also works as expected sounds reasonable to me too, I'll look into adding an additional assertion or test case for that.

Let me know if there is anything else in this PR that you would like to discuss further, or something that may require more changes or investigation.

Reply:
@Rob--W 

Actually I have to correct myself:
the generated UMD module doesn't export anything and so it can't be used as `import ... from './browser-polyfill.js'` and neither as `const ... = await import("./browser-polyfill")` because the returning module is going to be empty.

We could generate a separate "native" ES6 module which does export the API object, but I'm not sure that it is actually doable to mix the two together, because the export statement has to be on the top level scope and not inside the wrapper function scope.

I think that it may be more reasonable to defer looking more into that (generating a separate "native" ES6 module), because this is currently blocking the upgrade of this repo to a more recent babel version and I would really prefer that to be fixed as soon as we can.

The current version does pass all the existing test case, and so it should already be working as the current one, plus fixing #202 

In my opinion this is enough for now.

Reply:
What you just explained sounds reasonable. Let's fix #202 and not worry about `browser` export for now.